.github/workflows/ofrak-type-ci.yml

```diff
@@ -0,0 +1,619 @@
+      - 'ofrak_patch_maker/**'
+      - 'pytest_ofrak/**'
+      - 'ofrak_core/**'
+      - 'frontend/**'
+      - '.github/workflows/ofrak-type-ci.yml'
+  push:
+    branches:
+      - feature/windows_tests
+    #      - master
+#    paths:
+#      - 'ofrak_type/**'
+#      - 'ofrak_io/**'
+#      - '.github/workflows/ofrak-type-ci.yml'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash  # NOTE: On Windows, this uses Git Bash which provides Unix compatibility (for example for ofrak-io's StreamCapture
+
+jobs:
+  build-ofrak-type-sdist:
+    name: build ofrak-type sdist
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install build tools
+        run: |
+          python -m pip install -r pip-requirements.txt
+          pip install build
+
+      - name: Build OFRAK Type source distribution
+        run: |
+          cd ofrak_type
+          python -m build --sdist --outdir ../dist-type
+
+      - name: List built packages
+        run: |
+          echo "Built OFRAK Type packages:"
+          ls -la dist-type/
+
+      - name: Upload source distribution
+        uses: actions/upload-artifact@v4
+        with:
+          name: ofrak-type-sdist
+          path: dist-type/
+          retention-days: 30
+
+  test-ofrak-type:
+    name: Test OFRAK Type on ${{ matrix.os }} with Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    needs: build-ofrak-type-sdist
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
+        
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Download source distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: ofrak-type-sdist
+          path: dist-type/
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+
+      - name: Create virtual environment
+        run: |
+          python -m venv venv
+        
+      - name: Activate virtual environment (Linux/macOS)
+        if: runner.os != 'Windows'
+        run: |
+          source venv/bin/activate
+          echo "VIRTUAL_ENV=$VIRTUAL_ENV" >> $GITHUB_ENV
+          echo "$VIRTUAL_ENV/bin" >> $GITHUB_PATH
+          
+      - name: Activate virtual environment (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          venv\Scripts\activate
+          echo "VIRTUAL_ENV=$env:VIRTUAL_ENV" >> $env:GITHUB_ENV
+          echo "$env:VIRTUAL_ENV\Scripts" >> $env:GITHUB_PATH
+        shell: pwsh
```

Comment:
What is the benefit of using a virtual environment inside of the CI environment?